### PR TITLE
KAFKA-5368: Add test for skipped-records metric

### PR DESCRIPTION
**Pull request by mjsax**
- adding a missing test for KAFKA-5368

### CODE REVIEW

**Comment by mjsax** [comment]
Call for review @bbejeck @dguy @guozhangwang

**Comment by bbejeck** [comment]
@mjsax,  test failure is related

**Comment by mjsax** [comment]
Updated.

File: streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java [hunk 1]
```diff
@@ -825,6 +828,51 @@ public boolean conditionMet() {
...
+
+        final MockConsumer<byte[], byte[]> mockConsumer = (MockConsumer<byte[], byte[]>) thread.consumer;
+        mockConsumer.assign(Collections.singleton(t1p1));
+        mockConsumer.updateBeginningOffsets(Collections.singletonMap(t1p1, 0L));
+
+        final MetricName skippedTotalMetric = metrics.metricName("skipped-records-total", "stream-metrics", Collections.singletonMap("client-id", thread.getName()));
+        assertEquals(0.0, metrics.metric(skippedTotalMetric).metricValue());
+
+        long offset = -1;
+        mockConsumer.addRecord(new ConsumerRecord<>(t1p1.topic(), t1p1.partition(), ++offset, -1, TimestampType.CREATE_TIME, -1, -1, -1, new byte[0], new byte[0]));
```
Comment (bbejeck) [inline streams/src/test/java/org/apache/kafka/streams/processor/internals/StreamThreadTest.java:858]: maybe add a private method for adding records to `mockConsumer` for here and below?

**Comment by bbejeck** [comment]
Thanks for updating.  Just one minor comment, otherwise LGTM.

**Comment by bbejeck** [comment]
restest this please

**Comment by guozhangwang** [comment]
LGTM. Merged to trunk.

**Pull request merged**